azurerm/internal/services/datafactory/data_factory.go

```diff
@@ -403,3 +419,91 @@ func flattenDataFactoryDatasetAzureBlobFSLocation(input *datafactory.AzureBlobFS
 		},
 	}
 }
+func flattenDataFactoryDatasetSFTPLocation(input *datafactory.SftpLocation) []interface{} {
+	if input == nil {
+		return nil
```

Comment:
in general, when input is nil, we would return `[]interface{}{}` rather than nil

Reply:
I agree, but don't know how to solve this:
```
cannot use ([]interface{} literal) (value of type []interface{}) as datafactory.BasicDatasetLocation value in return statement: missing method AsAmazonS3CompatibleLocationcompilerInvalidIfaceAssign
```